src/edu/csus/ecs/pc2/clics/API202306/OrganizationService.java

```diff
@@ -0,0 +1,137 @@
+     * @return a {@link Response} object containing the model organizations in JSON form
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getOrganizations(@PathParam("contestId") String contestId) {
+
+        // keep track of which ones are being used for this contest
+        HashSet<String> orgSet = new HashSet<String>();
+
+        // check contest id
+        if(contestId.equals(model.getContestIdentifier()) == false) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        ArrayList<CLICSOrganization> olist = new ArrayList<CLICSOrganization>();
+
+        LoadAccounts.loadInstitutions(model);
+
+        // make up a list of CLICSOrganizations in use.
+        for(Account account: model.getAccounts()) {
+            if (account.getClientId().getClientType().equals(ClientType.Type.TEAM) && !account.getInstitutionCode().equals(Constants.DEFAULT_INSTITUTIONCODE)) {
+                String orgId = IJSONTool.getOrganizationId(account);
+                if(orgSet.add(orgId)) {
+                    String [] orgFields = ICPCTSVLoader.getInstitutionNames(orgId);
+                    if(orgFields != null) {
+                        olist.add(new CLICSOrganization(orgId, account, orgFields));
+                    }
+                }
+            }
+        }
+        try {
+            ObjectMapper mapper = JSONUtilities.getObjectMapper();
+            String json = mapper.writeValueAsString(olist);
+            return Response.ok(json, MediaType.APPLICATION_JSON).build();
+        } catch (Exception e) {
+            return Response.status(Status.INTERNAL_SERVER_ERROR).entity("Error creating JSON for organizations " + e.getMessage()).build();
+        }
+    }
+
+    @GET
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Path("{organizationId}/")
+    public Response getOrganization(@PathParam("contestId") String contestId, @PathParam("organizationId") String organizationId) {
+
+        // check contest id
+        if(contestId.equals(model.getContestIdentifier()) == true) {
+            LoadAccounts.loadInstitutions(model);
+
+            String [] orgFields = ICPCTSVLoader.getInstitutionNames(organizationId);
+            if(orgFields != null) {
+                // find a team who belongs to this organization
+                for(Account account: model.getAccounts()) {
+                    if (account.getClientId().getClientType().equals(ClientType.Type.TEAM) && IJSONTool.getOrganizationId(account).equals(organizationId)) {
+                        return Response.ok(new CLICSOrganization(organizationId, account, orgFields).toJSON(), MediaType.APPLICATION_JSON).build();
+                    }
+                }
+            }
+        }
+        return Response.status(Response.Status.NOT_FOUND).build();
+    }
+
+    /**
+     * Retrieve access information about this endpoint for the supplied user's security context
+     *
+     * @param sc User's security information
```

Comment:
Unused parameter `contest` should be documented.

Reply:
Fixed in next push (comment added)